pinot-controller/src/test/java/org/apache/pinot/controller/api/PinotSegmentsMetadataTest.java

```diff
@@ -113,7 +112,9 @@ private String serverName(int counter) {
   @AfterClass
   public void tearDown() {
     for (Map.Entry<String, SegmentsServerMock> fakeServerEntry : _serverMap.entrySet()) {
-      fakeServerEntry.getValue()._httpServer.stop(0);
+      SegmentsServerMock server = fakeServerEntry.getValue();
+      server.stop();
+      server._httpServer.stop(0);
```

Comment:
Is it possible to make this call inside the stop method of `SegmentsServerMock`, so that you don't have to explicitly call it everywhere?